### ISSUE TITLE
How can I visualize the input and output data?

### ISSUE THREAD

**Issue by 7675t** [Hippogriff/CSGNet#2]
I can train the network as README says.
Next, I want to see how good it predicts the output from a input.
Can you provide a instruction how can we do that?

**Comment by Hippogriff** [comment]
You can look at the test_synthetic.py file for testing the model. You can also also use visualize_expressions.py script to visualize the induced programs in the form of images.

**Comment by 7675t** [comment]
Thank you, it much helps to visualize the test results. I can close this issue. Also see my PR.

Can you consider to append a simple command which just apply the network to specified single image, visualize the predicted image and CSG expression? I think it is very helpful to know the concept of your work out of the box.

Thanks,

**Comment by Hippogriff** [comment]
Thank you for the PR but it doesn't seem necessary, because visualize_expressions.py can be used by the user by slight modification.

> Can you consider to append a simple command which just apply the network to specified single image, visualize the predicted image and CSG expression? I think it is very helpful to know the concept of your work out of the box.

I will add a small script.

**Comment by adrelino** [comment]
I still cannot find the script that just takes a 64x64 binary image (and the provided pre-trained model) as input and outputs a program code (that generates a similar looking image based on camfer or iou distance) for that image.

`visualize_expressions.py` just takes an expression and generates the ground-truth 64x64 image.

`test_synthetic.py` takes the pre-trained model and also the whole synthetic data-set expressions, somehow selects a validation subset and computes the mean chamfer distance metric over all program sizes. I guess that from this file, such a script that just takes an image and outputs the program could be extracted, but I am a little confused about these steps:
https://github.com/Hippogriff/CSGNet/blob/34e96210182692822ad6c7ff7b42dd1717157cfb/test_synthetic.py#L104-L120

When running inference, we should not need ground-truth labels, but then why do we need to extract one-hot-labels from the ground truth labels before feeding it to `test_output = imitate_net.test([data, one_hot_labels, max_len])`? 
How would I proceed if instead of using the generator `data_, labels = next(test_gen_objs[k])`, I just input my own image (batch), let's say a logo for which I don't even have ground truth labels?

Thanks for your help!

**Comment by adrelino** [comment]
I found out that `labels` is actually unused and just filled with 0, so can be ignore.

In my fork, I refactored the code a bit so that visualize_expressions now also returns the output of the pre-trained network in addition to the groundtruth. Can be found here:

https://github.com/adrelino/CSGNet/blob/master/visualize_expressions.py

**Comment by Hippogriff** [comment]
@adrelino Thanx for the script. The labels are unused in test mode.
